enhancements/cluster-logging/forward_to_azure_log_analytics.md

```diff
@@ -0,0 +1,182 @@
+    type: azureMonitor
+    azureMonitor:
+      customerId: "myCustomerID"
+      logType: "MyLogType"
+    secret:
+       name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+    outputRefs: [AzureLogAnalytics]
+```
+
+#### I want to forward logs to Azure Monitor: Log Analytics and associate the logs with some Azure resource
+```yaml
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+  - name: AzureLogAnalytics
+    type: azureMonitor
+    azureMonitor:
+      customerId: "myCustomerID"
+      logType: "MyLogType"
+      azureResourceId: "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/otherResourceGroup/examplestorage"
+    secret:
+       name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+    outputRefs: [AzureLogAnalytics]
+```
+
+#### I want to forward logs to Azure Monitor: Log Analytics in alternative host for dedicated Azure regions.
+```yaml
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+  - name: AzureLogAnalytics
+    type: azureMonitor
+    azureMonitor:
+      customerId: "myCustomerID"
+      logType: "MyLogType"
+      host: "ods.opinsights.azure.acme"
+    secret:
+       name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+    outputRefs: [AzureLogAnalytics]
+```
+
+### Implementation Details [optional]
+Use the [Vector plugin](https://vector.dev/docs/reference/configuration/sinks/azure_monitor_logs) to publish log events
+to the Azure Monitor Logs service
+Plugin configuration settings:
+
+The customer_id, log_type and shared_key fields are required, while the azure_resource_id and host field is optional.
+
+* customer_id : The unique identifier for the Log Analytics workspace.
+* log_type : Specify the record type of the data that's being submitted. It can contain only letters, numbers, and the underscore (_) character, and it can't exceed 100 characters. Each request to the Data Collector API must include a LogType with the name for the record type. The suffix _CL is automatically appended to the name you enter to distinguish it from other log types as a custom log. For example, if you enter the name MyNewRecordType, Azure Monitor creates a record with the type MyNewRecordType_CL. This helps ensure that there are no conflicts between user-created type names and those shipped in current or future Microsoft solutions.
+* shared_key : The primary or the secondary key for the Log Analytics workspace. Must be provided with secret
+* azure_resource_id : The resource ID of the Azure resource that the data should be associated with. It populates the _ResourceId property and allows the data to be included in resource-context queries. If this field isn't specified, the data won't be included in resource-context queries.
+* host: Azure China differs from Azure global, so service endpoints must be changed. Default: ods.opinsights.azure.com
```

Comment:
How is this different from "URL"?  How does the configuration find the right endpoint?  Is that something the client discovers with the id and other?

Reply:
In Vector plugin, URL (endpoint) build in this way: 
```
 let url = format!(
            "https://{}.{}{}?api-version={}",
            config.customer_id, config.host, RESOURCE, API_VERSION
        );
```
where: 
const RESOURCE: &str = "/api/logs";
const API_VERSION: &str = "2016-04-01";
if config.host not provided :
```
fn default_host() -> String {
    "ods.opinsights.azure.com".into()
}
```
Perhaps we can derive the URL parameter from OutputSpec. However, a potential issue arises as the current format is not in the standard URL form. This could lead to misunderstandings in the documentation, especially if the parameter is referred to as 'URL' instead of 'HOST'.